Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -261,6 +271,96 @@ theorem _root_.IsCentralScalar.isLinearTopology_iff [Module Rᵐᵒᵖ M] [IsCen
   · exact mk_of_hasBasis' Rᵐᵒᵖ (IsLinearTopology.hasBasis_submodule R)
       fun S r m hm ↦ unop_smul_eq_smul r m ▸ S.smul_mem _ hm
 
+variable (R M) in
+lemma nonarchimedeanAddGroup [IsTopologicalAddGroup M] [IsLinearTopology R M] :
+    NonarchimedeanAddGroup M := by
+  refine ⟨fun U hU ↦ ?_⟩
+  obtain ⟨N, hN, hNU⟩ := (hasBasis_submodule R).mem_iff.mp hU
+  exact ⟨⟨N.toAddSubgroup, N.toAddSubgroup.isOpen_of_mem_nhds hN⟩, hNU⟩
+
+lemma _root_.isLinearTopology_int_iff_nonarchimedeanAddGroup [IsTopologicalAddGroup M] :
+    IsLinearTopology ℤ M ↔ NonarchimedeanAddGroup M := by
+  refine ⟨fun _ ↦ IsLinearTopology.nonarchimedeanAddGroup ℤ M, fun _ ↦ ?_⟩
+  refine isLinearTopology_iff_exists_submodule_isOpen_subset.mpr fun s hs h0s ↦ ?_
+  obtain ⟨V, hV⟩ := NonarchimedeanAddGroup.is_nonarchimedean s (hs.mem_nhds h0s)
+  exact ⟨V.1.toIntSubmodule, V.2, hV⟩
```

Comment:
I would prefer to avoid the lemma `isLinearTopology_iff_exists_submodule_isOpen_subset` which looks un-idiomatic to me.
(suggestion that can be golfed a little bit.)

```suggestion
  simp only [isLinearTopology_iff_hasBasis_submodule, hasBasis_iff]
  intro s
  refine ⟨fun h0s ↦ ?_, fun ⟨V, hV0, hVs⟩ ↦ mem_of_superset hV0 hVs⟩
  obtain ⟨V, hV⟩ := NonarchimedeanAddGroup.is_nonarchimedean s h0s
  exact ⟨V.1.toIntSubmodule, (isOpen_iff_mem_nhds.mp V.2) 0 V.zero_mem, hV⟩
```

Reply:
The real way to trivialize this lemma and the one above would be to have a proper description of [NonarchimedeanGroup](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Topology/Algebra/Nonarchimedean/Basic.html#NonarchimedeanGroup) in terms of filter bases, which simply doesn't exist at the time being!

Adding the lemmas would be relatively easy, but it might be worth changing the definition as well... although I'm not sure how disturbing this would be for the FLT team...